core/capabilities/ccip/oraclecreator/plugin.go

```diff
@@ -208,18 +210,25 @@ func (i *pluginOracleCreator) createFactoryAndTransmitter(
 	var factory ocr3types.ReportingPluginFactory[[]byte]
 	var transmitter ocr3types.ContractTransmitter[[]byte]
 	if config.Config.PluginType == uint8(cctypes.PluginTypeCCIPCommit) {
+		if !i.peerWrapper.IsStarted() {
+			return nil, nil, fmt.Errorf("peer wrapper is not started")
+		}
+		rmnPeerClient := rmn.NewPeerClient(i.peerWrapper.GenericNetEndpointFactory)
```

Comment:
GenericNetEndpointFactory provided to the PeerClient to further initialize peer connections on demand. Since at this point the current oracle might not be able to read the destination chain and we want RMNRemote contract data, which live on the destination chain.